packages/coil-extension/src/background/services/BackgroundScript.ts

```diff
@@ -175,6 +176,53 @@ export class BackgroundScript {
     })
   }
 
+  private modifyPaymentHeader(details: any) {
+    // Check if the header has been defined already
+
+    let found = false
+    const paymentHeader = 'accept-payment'
+    const paymentPreference = 'webmon/*;q=.8'
+
+    // it's possible that another extension will be adding this header
+    // maybe an adblocker wants to add ads/*;q=0,ads/acceptable;q=.5
+    for (const header of details.requestHeaders) {
+      if (header.name.toLowerCase() === paymentHeader) {
+        found = true
+        if (
+          header.value &&
+          header.value.toLowerCase().indexOf('webmon') === -1
+        ) {
+          // already exists, add webmon if it's not already there
+          header.value = `${header.value},${paymentPreference}`
```

Comment:
Pity it *seems* can't do this with declarativeNetRequest

Reply:
Oh good call. I looked into it and this is possible. I'll close this PR and open another one.